src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java

```diff
@@ -0,0 +1,170 @@
+import org.apache.commons.lang3.StringUtils;
+import org.folio.ActionProfile;
+import org.folio.DataImportEventPayload;
+import org.folio.MappingProfile;
+import org.folio.inventory.common.Context;
+import org.folio.inventory.common.api.request.PagingParameters;
+import org.folio.inventory.dataimport.handlers.matching.util.EventHandlingUtil;
+import org.folio.inventory.domain.items.CirculationNote;
+import org.folio.inventory.domain.items.Item;
+import org.folio.inventory.domain.items.ItemCollection;
+import org.folio.inventory.domain.items.ItemStatusName;
+import org.folio.inventory.storage.Storage;
+import org.folio.inventory.support.CqlHelper;
+import org.folio.inventory.support.ItemUtil;
+import org.folio.inventory.support.JsonHelper;
+import org.folio.processing.events.services.handler.EventHandler;
+import org.folio.processing.exceptions.EventProcessingException;
+import org.folio.processing.mapping.MappingManager;
+import org.folio.rest.jaxrs.model.EntityType;
+
+import java.io.UnsupportedEncodingException;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static org.folio.ActionProfile.Action.CREATE;
+import static org.folio.ActionProfile.FolioRecord.ITEM;
+import static org.folio.rest.jaxrs.model.ProfileSnapshotWrapper.ContentType.ACTION_PROFILE;
+
+public class CreateItemEventHandler implements EventHandler {
+
+  public static final String ITEM_CREATED_EVENT_TYPE = "DI_INVENTORY_ITEM_CREATED";
+  private static final String PAYLOAD_HAS_NO_DATA_MSG = "Failed to handle event payload, cause event payload context does not contain MARC_BIBLIOGRAPHIC data";
+
+  private static final Logger LOG = LoggerFactory.getLogger(CreateItemEventHandler.class);
+
+  private final DateTimeFormatter dateTimeFormatter =
+    DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ").withZone(ZoneOffset.UTC);
+
+  private final List<String> requiredFields = Arrays.asList("status.name", "materialType.id", "permanentLoanType.id", "holdingsRecordId");
+
+  private Storage storage;
+
+  public CreateItemEventHandler(Storage storage) {
+    this.storage = storage;
+  }
+
+  @Override
+  public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload dataImportEventPayload) {
+    CompletableFuture<DataImportEventPayload> future = new CompletableFuture<>();
+    try {
+      HashMap<String, String> payloadContext = dataImportEventPayload.getContext();
+      if (payloadContext == null || StringUtils.isBlank(payloadContext.get(EntityType.MARC_BIBLIOGRAPHIC.value()))) {
+        LOG.error(PAYLOAD_HAS_NO_DATA_MSG);
+        future.completeExceptionally(new EventProcessingException(PAYLOAD_HAS_NO_DATA_MSG));
+        return future;
+      }
```

Comment:
@RuslanLavrov please add some tests for this case

Reply:
already added